src/libraries/System.Private.CoreLib/src/System/Numerics/Matrix4x4.cs

```diff
@@ -1803,188 +1803,270 @@ public static bool Decompose(Matrix4x4 matrix, out Vector3 scale, out Quaternion
         {
             bool result = true;
 
-            unsafe
+            float det;
+
+            Matrix4x4 matTemp = Identity;
+            Vector3 canonicalBasis_Row0 = Unsafe.As<float, Vector3>(ref matTemp.M11);
+            Vector3 canonicalBasis_Row1 = Unsafe.As<float, Vector3>(ref matTemp.M21);
+            Vector3 canonicalBasis_Row2 = Unsafe.As<float, Vector3>(ref matTemp.M31);
```

Comment:
This is a fairly expensive way to get what will end up being `Vector3.UnitX`, `Vector3.UnitY`, `Vector3.UnitZ`